packages/build-tools/src/steps/easFunctions.ts

```diff
@@ -46,10 +46,22 @@ export function getEasFunctions(ctx: CustomBuildContext): BuildFunction[] {
     createStartAndroidEmulatorBuildFunction(),
     createStartIosSimulatorBuildFunction(),
     createInstallMaestroBuildFunction(),
-    createGetCredentialsForBuildTriggeredByGithubIntegration(ctx),
+
     createInstallPodsBuildFunction(),
     createSendSlackMessageFunction(),
-    createResolveBuildConfigBuildFunction(ctx),
-    calculateEASUpdateRuntimeVersionFunction(),
+
+    calculateEASUpdateRuntimeVersionFunction(ctx),
   ];
+
+  if (ctx.hasBuildJob()) {
+    functions.push(
+      ...[
+        createFindAndUploadBuildArtifactsBuildFunction(ctx),
+        createResolveBuildConfigBuildFunction(ctx),
+        createGetCredentialsForBuildTriggeredByGithubIntegration(ctx),
+      ]
+    );
+  }
```

Comment:
Why did you pick these as BuildJob only and not the others? I mean I'm not saying it's wrong, just want to understand it better

Reply:
Of course! What I did is change the meaning of `Job` to include `Generic.Job`. This, obviously caused some TS errors, like "Property `type` does not exist [in `job`]". Then, I inspected each error and:
- if a function made sense in Generic Jobs context, I made it support it (by adding `'type' in job` and others)
- if a function did not make sense in Generic Jobs context (like these), I changed the expected context type to specify `BuildJob`. Then I couldn't pass a `Job` context to these functions, so I excluded them like so.

So, essentially, I'm excluding only the functions that don't work for `Generic.Job`. Maybe more functions don't make sense, here I'm only focusing on type layer, not "product layer".